cherrypy/lib/static.py

```diff
@@ -174,7 +176,22 @@ def _serve_fileobj(fileobj, content_type, content_length, debug=False):
                 cherrypy.log(message, 'TOOLS.STATIC')
             raise cherrypy.HTTPError(416, message)
 
-        if r:
+        is_if_range_in_past = False
+        if_range_header = request.headers.get('If-Range')
+
+        if if_range_header:
+            # Per RFC:
+            # The If-Range HTTP request header makes a range request
+            # conditional: if the condition is fulfilled, the range
+            # request will be issued and the server sends back
+            # a 206 Partial Content answer with
+            # the appropriate body. If the condition is not fulfilled,
+            # the full resource is sent back, with a 200 OK status.
+            # Ref: https://tools.ietf.org/html/rfc7233#section-3.2
+            if datetime(*parsedate(if_range_header)[:6]) < datetime.now():
```

Comment:
This will probably fail with E-Tag value:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Range

Reply:
I'll write a change for this. Do you think searching for exactly two double quotes will be enough for detecting ETAG vs Date? That's what I'm working with at the moment.

Reply:
I'd probably go for try/except

Reply:
And it should probably go to `httputil` module

Reply:
Migrated to httputil, and caught exception when there is an ETag instead of an httpdate. However, I treat the ETag as an always-fail condition as to my knowledge we don't support that yet? Correct me if I'm wrong.

Reply:
It's actually supported in `caching` tool: https://github.com/cherrypy/cherrypy/search?q=expires&unscoped_q=expires&type=Code

And it just occurred to me that maybe this check should be done there, not in this module.

There's also a tool called `expires` + there's tests for etag as well.

Could you please explore this?